Objective-C/Internal/CBLCollection+Internal.h

```diff
@@ -33,7 +35,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-@interface CBLCollection () <CBLRemovableListenerToken>
+@interface CBLCollection () <CBLRemovableListenerToken, CBLIndexableInternal>
```

Comment:
I don't see a reason to create a protocol for CBLIndexableInternal as CBLCollection will not be used as CBLIndexableInternal type. I think you can just add a method `indexesInfo` in CBLCollection() directly.